kitchen/bin/kitchen

```diff
@@ -0,0 +1,901 @@
+            self.__status = 400
+            self.__set_response(b'Missing request-id')
+        else:
+            with _async_state_lock:
+                req_data = _async_state.get(req_id, {})
+            if not req_data:
+                self.__status = 404
+            response_json = json.dumps(dict(status=self.__status, **req_data))
+            self.__headers['Content-Type'] = 'application/json'
+            self.__set_response(response_json.encode('utf-8'))
+
+    def __request_info(self):
+        """Build request-info endpoint dict data JSON response."""
+        info = { 'request-method': self.__method, 'uri': self.__path }
+        if len(self.headers) > 0:
+            info['headers'] = { k.lower(): v for k, v in self.headers.items() }
+        if self.__query is not None:
+            info['query-string'] = self.__query
+        return json.dumps(info, sort_keys=True)
+
+    def __set_response(self, response_bytes, response_length=None):
+        """Set response payload, and update the content length accordingly."""
+        self.__response_bytes = response_bytes
+        if response_length is None:
+            self.__response_length = len(response_bytes)
+        else:
+            self.__response_length = response_length
+        # Limit response length to the maximum supported size
+        self.__response_length = min(max_response_size, self.__response_length)
+
+    def __slurp_bytes(self, bytes_to_read):
+        """Consume (and throw away) data from the request body."""
+        self.logger().debug('Consuming {} bytes from request payload'.format(bytes_to_read))
+        while bytes_to_read > 0:
+            n = min(bytes_to_read, default_chunk_size)
+            bytes_to_read -= len(self.rfile.read(n))
+
+    def __state(self):
+        """Build kitchen-state endpoint dict data JSON response."""
+        with _async_state_lock, _counter_lock:
+            state = {
+                'async-requests': _async_state,
+                'pending-http-requests': _pending_http_requests,
+                'pending-ws-requests': _pending_ws_requests,
+                'total-http-requests': _total_http_requests,
+                'total-ws-requests': _total_ws_requests
+                }
+        self.__headers['Content-Type'] = 'application/json'
+        return json.dumps(state, sort_keys=True).encode('utf-8')
+
+class MultiThreadedServer(socketserver.ThreadingMixIn, http.server.HTTPServer):
+    """Serve each HTTP request on separate thread"""
+    pass
+
+class BasicAuthHandler():
+    """Functor for verifying Waiter BasicAuth credentials."""
+    def __init__(self, username, password):
+        b64_creds = base64.b64encode(':'.join([username, password]).encode('utf-8'))
+        self.expected_auth = 'Basic {}'.format(b64_creds.decode())
+    def __call__(self, auth_string):
+        return auth_string == self.expected_auth
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(description='Mock HTTP Service')
```

Comment:
"Mock" seems like the wrong description. Maybe "Testing"? Also, should we provide some more description about what this is used for?

Reply:
Description updated: *"A toy HTTP Service for testing the Waiter platform"*

If someone wants more detail than that, they can take a look at the README. I think people are more likely to look at the README than this help string anyway (especially since the kitchen command can run with no arguments, so the only way to get the help description is to explicitly ask for it).